MediaGalleryUi/Model/Directories/CreateByPath.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\Directories;
+
+use Magento\Cms\Model\Wysiwyg\Images\Storage;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Create folder by provided path
+ */
+class CreateByPath
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Storage
+     */
+    private $storage;
+
+    /**
+     * @param LoggerInterface $logger
+     * @param Storage $storage
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        Storage $storage
+    ) {
+        $this->logger = $logger;
+        $this->storage = $storage;
+    }
+
+    /**
+     * Create new directory by provided path
+     *
+     * @param string $path
+     * @param string $name
+     * @throws CouldNotSaveException
+     */
+    public function execute(string $path, string $name): void
+    {
+        try {
+            $this->storage->createDirectory($name, $this->storage->getCmsWysiwygImages()->getStorageRoot() . $path);
```

Comment:
Potentially it's possible to create a directory outside of media directory by sending path with '..' from frontend - let's try to filter it here and ensure all the directory method in framework go through `getRealPathSafety` filter (PR to magento2 required for this one)